pkg/terminal/proxy.go

```diff
@@ -89,19 +89,19 @@ func (p *Proxy) HandleProxy(user *auth.User, w http.ResponseWriter, r *http.Requ
 		return
 	}
 
-	enabledForUser, err := p.checkUserPermissions(user.Token)
+	ok, namespace, workspaceName, path := stripTerminalAPIPrefix(r.URL.Path)
+	if !ok {
+		http.NotFound(w, r)
+		return
+	}
+
+	enabledForUser, err := p.checkUserPermissions(user.Token, namespace)
```

Comment:
Should we sync checkUserPermissions with the frontend part which has
```
if (group === 'system:cluster-admins') {
```

Reply:
Technically yes, but this isn't as easy on the backend as the user object doesn't have groups AFAICT.

@spadgett or @christianvogt Do you know how to do this? Essentially the flow could be: When the user tries to open a terminal living in the openshift-terminal namespace check if their user is in the cluster-admin group. If they are not then deny.

On the frontend, we seem to get the groups that the user is associated with but on the backend it's not as simple

Reply:
The logic in https://github.com/openshift/console/pull/7145/commits/59b3c77556b89748e8eda0878c9233a905c9e956 does not seem to be right but I may be wrong.
I assume it should be rewritten in the following way: https://github.com/JPinkney/console/compare/secured-namespaces...sleshchenko:secured-namespace-patch?expand=1